### ISSUE TITLE
Password-only Secret as StringCredential with plain-credentials

### ISSUE THREAD

**Issue by luka5** [openshift/jenkins-sync-plugin#216]
Hi everyone,

is there a reason, why a password-only (also called token) OpenShift Basic-Auth Secret is not mapped as a [`StringCredential`](https://github.com/jenkinsci/plain-credentials-plugin/blob/master/src/main/java/org/jenkinsci/plugins/plaincredentials/StringCredentials.java) using the `plain-credentials-plugin` ?

Currently, the username is mandatory for this plugin but not for the OpenShift Basic-Auth Secret. Hence, a Basic-Auth Secret without a username leads to a [`Warning: Invalid secret data ...`](https://github.com/openshift/jenkins-sync-plugin/blob/master/src/main/java/io/fabric8/jenkins/openshiftsync/CredentialsUtils.java#L296) with `usernameData is null`.

Would you also choose the `plain-credentials-plugin` to provide password-only credentials?

Thanks!

**Comment by gabemontero** [comment]
Hey @luka5 

Yeah in general this plugin just patterned itself after the k8s plugin when it was created.

Based on your prompting here, I took a peek at its more recent code, and see it has started deprecating their `BaseStandardCredentials` stuff in favor of `StringCredentials`, etc.  Appears to be the new "Jenkins way" here.

Per my team's process, I have opened https://trello.com/c/rum3SCfs/1538-follow-k8s-plugin-lead-switch-to-plain-credentials-plugin-for-all-our-plugins-jenkinsintegration to track this requirement and will close this card out.

Fell free to join our trello board at https://trello.com/b/wqe9UaaZ/developer-experience if you want to track that card, chime in on use cases, etc.  .. though we can continue to exchange additional messages here if that is your preference.

One side bar, to possibly sync up on further ... not sure what you mean exactly by `OpenShift Basic-Auth Secret`.

If you look at the credential types we register with jenkins with a relatively recent version of this plugin, and try to add a credential which this plugin processes, the textual title is `OpenShift Token for OpenShift Sync Plugin`

Any other credential type with `OpenShift` in the name/kind/descrption is a credential type offered by another plugin.
